### ISSUE TITLE
- error TS2304: Cannot find name 'USBDevice'.

### ISSUE THREAD

**Issue by rius1212** [bebo925/ng-thermal-print#37]
hi bebo925 , i got error when try to start project using ng serve. i already try to use this command npm install @types/w3c-web-usb --only=dev but i still got error like what i put in the title issue. any idea what i need to do ?

ERROR in node_modules/ng-thermal-print/lib/drivers/UsbDriver.d.ts:16:30 - error TS2304: Cannot find name 'USBDevice'.

16     requestUsb(): Observable<USBDevice>;
                                ~~~~~~~~~
node_modules/ng-thermal-print/lib/drivers/UsbDriver.d.ts:1:23 - error TS2688: Cannot find type definition file for 'w3c-web-usb'.

1 /// <reference types="w3c-web-usb" />

**Comment by jooskesters** [comment]
having the same issue here

**Comment by cooper1388** [comment]
I have the same error, have you been able to solve it?

**Comment by jooskesters** [comment]
No... I suggest you step away from these libraries and use java.. With java you can copy some printing code from stackoverflow and it will work most of the time. You can then create a background service from your java program that handles the printing jobs ..

This is how I fixed it. Works fine, just some more work then expected when I put my faith in these stupid npm libraries

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by ledger32570** [comment]
The worst developer experience ever! Every single issue I encounter lands me on an unresolved or prematurely closed GitHub issue!
